x-pack/plugins/ingest_manager/server/routes/agent/upgrade_handler.ts

```diff
@@ -35,15 +37,25 @@ export const postAgentUpgradeHandler: RequestHandler<
       },
     });
   }
-  const agent = await soClient.get<AgentSOAttributes>(
+  const agentSO = await soClient.get<AgentSOAttributes>(
     AGENT_SAVED_OBJECT_TYPE,
     request.params.agentId
   );
-  if (agent.attributes.unenrollment_started_at || agent.attributes.unenrolled_at) {
+  if (agentSO.attributes.unenrollment_started_at || agentSO.attributes.unenrolled_at) {
```

Comment:
we also have the `.active = false` property when an agent is unenrolled that I will trust more than these two field

Reply:
I used this because it was used [here to get unenrolled or unenrolling](https://github.com/elastic/kibana/blob/master/x-pack/plugins/ingest_manager/server/services/agents/unenroll.ts#L47) and because active doesn't get set to false until after unenrollment.  Going to commit but will review when we do the status refactor.